htlcswitch/packet.go

```diff
@@ -0,0 +1,61 @@
+package htlcswitch
+
+import (
+	"crypto/sha256"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+	"github.com/roasbeef/btcd/wire"
+)
+
+// htlcPacket is a wrapper around an lnwire message which adds, times out, or
+// settles an active HTLC. The dest field denotes the name of the interface to
+// forward this htlcPacket on.
+type htlcPacket struct {
+	// payHash payment hash of htlc request.
+	payHash [sha256.Size]byte
+
+	// dest is the next peer in hlc path.
+	dest routing.HopID
+
+	// source is a previous channel to which htlc was applied.
+	source wire.OutPoint
+
+	// htlc lnwire message type of which depends on switch request type.
+	htlc lnwire.Message
+}
+
+// newUpdatePacket creates htlc swicth update packet which encapsulates the
+// add htlc request and additional information for proper forwarding over
+// htlc switch.
+func newUpdatePacket(source wire.OutPoint, dest routing.HopID,
```

Comment:
Naming suggestion: `newAddPacket`, and `newSettlePacket` for the function below.